pkg/controller/utils/elasticsearch.go

```diff
@@ -62,3 +104,266 @@ func GetElasticsearchClusterConfig(ctx context.Context, cli client.Client) (*ren
 
 	return render.NewElasticsearchClusterConfigFromConfigMap(configMap)
 }
+
+type ElasticClient interface {
+	SetILMPolicies(client.Client, context.Context, *operatorv1.LogStorage) error
+}
+
+type esClient struct {
+	client *elastic.Client
+	lock   *sync.Mutex
+}
+
+func NewElasticClient() ElasticClient {
+	return &esClient{}
+}
+
+// SetILMPolicies creates ILM policies for each timeseries based index using the retention period and storage size in LogStorage
+func (es *esClient) SetILMPolicies(client client.Client, ctx context.Context, ls *operatorv1.LogStorage) error {
+	if err := es.createElasticClient(client, ctx); err != nil {
+		return err
+	}
+	policyList := es.listILMPolicies(ls)
+	return es.createOrUpdatePolicies(ctx, policyList)
+}
+
+func (es *esClient) createElasticClient(client client.Client, ctx context.Context) error {
+	if es.client == nil {
+		es.lock.Lock()
+		defer es.lock.Unlock()
+		user, password, root, err := getClientCredentials(client, ctx)
+		if err != nil {
+			return err
+		}
+		h := &http.Client{
+			Transport: &http.Transport{TLSClientConfig: &tls.Config{RootCAs: root}},
+		}
+
+		options := []elastic.ClientOptionFunc{
+			elastic.SetURL(render.ElasticsearchHTTPSEndpoint),
+			elastic.SetHttpClient(h),
+			elastic.SetErrorLog(logrus.StandardLogger()),
```

Comment:
Interface that `elastic.SetErrorLog()` expects is different from the one available in `sigs.k8s.io/controller-runtime/pkg/log` package, so had to use logrus package.

Reply:
Hmm it only needs to meet the interface
```
type Logger interface {        
    Printf(format string, v ...interface{})
} 
```
I think we should wrap the logger we use with something like
```
type logrWrappedESLogger struct {
  l logr.Logger
}
func (l logrWrappedESLogger) Printf(format string, v ...interface{}) {
  l.l.Error(nil, fmt.Printf(format, v...))
}
```
I've used the Error function in the wrapped call because we're passing it to SetErrorLog.
WDYT?
That was we ensure all our logs are in the same format and have the same metadata.